### PR TITLE
Genecard and Stringdb UI integration

### PR DESCRIPTION
**Pull request by arpitaagrawal**
@chrismattmann Hello Professor, committing the changes for Genecard and StringDB UI integration. Please review the same. Thanks!

### CODE REVIEW

File: shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js [hunk 1]
```diff
@@ -210,6 +210,98 @@ function searchSelectedText(selectedText)
...
+					var currentEngine3 = engine;
+					$.ajax({
+						headers : {
+							"Content-Type" : "text/plain"
+						},
+						url: searchPreferences[engine]["restURL"], 
+						method:"put",
+						data: selectedText,
+						success:function( responseObjects){
+							//$(".loading-img").remove();
```
Comment (chrismattmann) [inline shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js:212]: please remove this commented line out

File: shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js [hunk 2]
```diff
@@ -210,6 +210,98 @@ function searchSelectedText(selectedText)
...
+							
+							var searchResultGeneCard = "<ul class='nav nav-list'><li style='list-style: none; display: inline'>";
+							
+							for (var key in responseObjects) {
+								geneCardSearchLength++;
+								geneCardResult += "<ul class='geneCard'><li><strong>Symbol:</strong>"+responseObjects[key]["Symbol"] +"</li><li><strong>Description:</strong>" + 
+								responseObjects[key]["Description"] + "</li><li><a class='results' href='#'>View Details...</a></li></ul> <br />";
+							}
+							$("#geneCard .badge").html("<span class='badge'>"+ geneCardSearchLength +"</span>");
+//							$(".searchTab" + openFileIndex).append("<li role='presentation'><a href='#" + currentEngine3  + openFileIndex + "' data-toggle='tab'>" + currentEngine3 + " (" + wikiSearchLength + ")</a></li>");
```
Comment (chrismattmann) [inline shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js]: remove

File: shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js [hunk 3]
```diff
@@ -210,6 +210,98 @@ function searchSelectedText(selectedText)
...
+//							var searchResultWiki = "<div role='tabpanel' class='tab-pane ' id='" + currentEngine3  + openFileIndex + "'><ul class='searchList'>" + searchResultWiki;
```
Comment (chrismattmann) [inline shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js]: remove

File: shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js (hunk 2 above)
Comment (chrismattmann) [inline shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js]: remove

File: shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js [hunk 4]
```diff
@@ -210,6 +210,98 @@ function searchSelectedText(selectedText)
...
+						error: function( e){
+							//sometimes even correct results error out because response is a string instead of an object.
+							$(".searchTab" + openFileIndex).append("<li role='presentation'><a href='#" + currentEngine3  + openFileIndex + "' data-toggle='tab'>" + currentEngine3 + "(Error)</a></li>");
+							var searchResultWiki = "<div role='tabpanel' class='tab-pane ' id='" + currentEngine3  + openFileIndex + "'><ul class='searchList'>";
+							searchResultWiki += "<label> Wikipedia results couldn't be fetched correctly. We are working on this issue.</label>";
+							$(".searchContent" + openFileIndex).append( searchResultWiki);
+
+						}
+					}).fail( function(){
+						//$(".searchContent"+ openFileIndex).html( "<label class='alert alert-danger'> An error has occurred. Please refresh the page and try again.</label>");
```
Comment (chrismattmann) [inline shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js]: remove or replace with an actual fail function

File: shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js [hunk 5]
```diff
@@ -210,6 +210,98 @@ function searchSelectedText(selectedText)
...
+					var currentEngine3 = engine;
+					$.ajax({
+						headers : {
+							"Content-Type" : "text/plain"
+						},
+						url: searchPreferences[engine]["restURL"], 
+						method:"put",
+						data: selectedText,
+						success:function( responseObjects){
+							//$(".loading-img").remove();
```
Comment (chrismattmann) [inline shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js]: same , remove

File: shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js [hunk 6]
```diff
@@ -210,6 +210,98 @@ function searchSelectedText(selectedText)
...
+							
+							var searchResultStringDB = "<ul class='nav nav-list'><li style='list-style: none; display: inline'>";
+							
+							for (var key in responseObjects) {
+								stringDBSearchLength++;
+								searchResultStringDB += "<ul class='stringDB'><li><strong>Organism:</strong>"+responseObjects[key]["taxonName"] +"</li><li><strong>Description:</strong>" + 
+								responseObjects[key]["description"] + "</li><li><a class='results' href='"+responseObjects[key]["link"]+"'>View Details...</a></li></ul> <br />";
+							}
+							$("#stringDB .badge").html("<span class='badge'>"+ stringDBSearchLength +"</span>");
+//							$(".searchTab" + openFileIndex).append("<li role='presentation'><a href='#" + currentEngine3  + openFileIndex + "' data-toggle='tab'>" + currentEngine3 + " (" + wikiSearchLength + ")</a></li>");
```
Comment (chrismattmann) [inline shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js]: remove next 3 lines

File: shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js [hunk 7]
```diff
@@ -210,6 +210,98 @@ function searchSelectedText(selectedText)
...
+						error: function( e){
+							//sometimes even correct results error out because response is a string instead of an object.
+							$(".searchTab" + openFileIndex).append("<li role='presentation'><a href='#" + currentEngine3  + openFileIndex + "' data-toggle='tab'>" + currentEngine3 + "(Error)</a></li>");
+							var searchResultWiki = "<div role='tabpanel' class='tab-pane ' id='" + currentEngine3  + openFileIndex + "'><ul class='searchList'>";
+							searchResultWiki += "<label> Wikipedia results couldn't be fetched correctly. We are working on this issue.</label>";
+							$(".searchContent" + openFileIndex).append( searchResultWiki);
+
+						}
+					}).fail( function(){
+						//$(".searchContent"+ openFileIndex).html( "<label class='alert alert-danger'> An error has occurred. Please refresh the page and try again.</label>");
```
Comment (chrismattmann) [inline shangridocs-webcomponents/src/main/java/org/shangridocs/webcomponents/search/SearchServicesPanel_v2.js]: actual fail function

**Comment by arpitaagrawal** [comment]
@chrismattmann Incorporated the changes! :)

**Comment by chrismattmann** [comment]
LGTM @arpitaagrawal ! merged.

**Pull request merged**